lib/Controller/UsersController.php

```diff
@@ -936,4 +1216,12 @@ private function isAdmin() {
 		}
 		return false;
 	}
+
+	private function logout() {
+		$loginToken = $this->request->getCookie('oc_token');
```

Comment:
not sure if all this is enough, especially when thinking of an impersonated user that gets logged out, which usually results in the admin being logged in again. please verify

Reply:
Need to test, I will update here.

Reply:
Hum. You are right. So I guess we can remove this method. Let the session continue. So in case if user really wants to login as new user, then let user manually close the session ( say by logout ) and then login as new user.